### PR TITLE
Use pytest to run each test

### PR DESCRIPTION
**Pull request by mwperina**
* Improve installation documentation
* Add the new test infrastructure for pytest
* Rewrite hirte service startup test in pytest
* Add test to check that agent is started and connected to controller

Signed-off-by: Martin Perina <mperina@redhat.com>

### CODE REVIEW

File: tests/tier0/hirte-service-startup/main.fmf [hunk 1]
```diff
@@ -3,4 +3,4 @@ summary: Test if hirte service started successfully
 tier: 0
 
 test: |
-  ./test.sh
\ No newline at end of file
+  pytest -svv
```
Comment (Yarboa) [inline tests/tier0/hirte-service-startup/main.fmf:6]: I would advise to add Regex variable,
to control the tests.

Currently, what ever found by pytest.ini will be run
Assume the same infrastructure is used in hirte pipeline.

Tests running over there will be different from tests AutoSD image for example

**Review by Yarboa — COMMENTED** [review]
Interesting approach
Why do you plan to use pytest? 
The benefit of tmt is actually just for running the tests,
Did i miss some thing here? 

One more issue is the reporting mechanism 
Assume you have 3 tests running in pytest,
They are visible as single test, it is not good for interaction with QMS systesms

File: tests/tier0/hirte-service-startup/main.fmf (hunk 1 above)
Comment (mwperina) [reply tests/tier0/hirte-service-startup/main.fmf:6]: Why? Here I want to perform actually 3 tests:

1. Check that hirte controller is running (and maybe respoding to requests)
2. Check that hirte-agent on node foo is running and successfully connected to to hirte (maybe even checking it's possible to fetch data from the node foo)
3. Check the same for hirte-agent on node bar

Of course other tests (for example proxy tests) will be located under different `tests/tier0/<PROXY TEST> directories`

File: tests/tier0/hirte-service-startup/test_service_startup.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import pytest
+
+from hirte_tests.common import fixtures
```
[the remark this answers is not in the document]
Comment (Yarboa) [reply tests/tier0/hirte-service-startup/test_service_startup.py:5]: Once I reapplied the plugin import
There are many podmanclient API errors

File: tests/tier0/hirte-service-startup/main.fmf (hunk 1 above)
Comment (Yarboa) [reply tests/tier0/hirte-service-startup/main.fmf:6]: You are correct,
The problem is, the 3 pytests are run under one tmt test
as a result tmt junit xml, will look like this 

Actually the report will look like that 
?xml version="1.0" ?>
<testsuites disabled="0" errors="0" failures="1" tests="1" time="0.0">
        <testsuite disabled="0" errors="0" failures="1" name="/plans/tier0" skipped="0" tests="1" time="0">
                <testcase name="/tests/tier0/hirte-service-startup">

but actuall you have 3 running tests

Comment (Yarboa) [reply tests/tier0/hirte-service-startup/main.fmf:6]: Please refer here 
https://github.com/oamg/convert2rhel/blob/main/tests/integration/tier0/config-file/main.fmf
He splits the tests in the main file

File: tests/tier0/hirte-service-startup/test_service_startup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mwperina) [reply tests/tier0/hirte-service-startup/test_service_startup.py]: With the last patch iteration it really works for me fine

File: tests/tier0/hirte-service-startup/main.fmf (hunk 1 above)
Comment (mwperina) [reply tests/tier0/hirte-service-startup/main.fmf:6]: No idea what QMS means.

If needed, we can add those description similar to above ...

File: scripts/tests-setup.sh [hunk 3]
```diff
@@ -1,11 +1,8 @@
...
-podman run -d --net hirte-test --rm -p $MGR_PORT:$MGR_PORT --name hirte-mgr \
-          $(podman images *hirte-image --format {{.ID}})
-podman run -d --net hirte-test --rm --name hirte-node-bar \
-          $(podman images *hirte-image --format {{.ID}})
-podman run -d --net hirte-test --rm --name hirte-node-foo \
-          $(podman images *hirte-image --format {{.ID}})
+podman network exists hirte-test
+if [ $? -ne 0]; then
+  podman network create --subnet $TEST_NET_RANGE hirte-test
+fi
```
Comment (Yarboa) [inline scripts/tests-setup.sh:8]: I would add here the, it is obviously need to be set up
Set up the REST API service

podman system service -t 0

this is why it was generating API errors here
https://github.com/mwperina/hirte-tests/pull/7#discussion_r1172790540

**Review by Yarboa — CHANGES_REQUESTED** [review]
LGTM, but there are short fixes

File: scripts/tests-setup.sh [hunk 4]
```diff
@@ -1,11 +1,8 @@
...
 
-podman network create --subnet $TEST_NET_RANGE hirte-test
-podman run -d --net hirte-test --rm -p $MGR_PORT:$MGR_PORT --name hirte-mgr \
-          $(podman images *hirte-image --format {{.ID}})
-podman run -d --net hirte-test --rm --name hirte-node-bar \
-          $(podman images *hirte-image --format {{.ID}})
-podman run -d --net hirte-test --rm --name hirte-node-foo \
-          $(podman images *hirte-image --format {{.ID}})
+podman network exists hirte-test
+if [ $? -ne 0]; then
```
[the remark this answers is not in the document]
Comment (mwperina) [reply scripts/tests-setup.sh]: Fixed

File: scripts/tests-setup.sh (hunk 3 above)
Comment (mwperina) [reply scripts/tests-setup.sh:8]: Why? This needs to be set up as a prerequisite. But unless I'm missing something it's required only on Fedora or CS9, on ubuntu it's enabled by default

Comment (Yarboa) [reply scripts/tests-setup.sh:8]: In Fedora it is required.
I assume check will not harm

**Review by Yarboa — APPROVED** [review]
LGTM,

**Pull request merged**